studio gql support docs (#5834)
aws-amplify/docs: src/pages/console/data/data-model.mdx

## Data modeling example

<Callout warning>

> **Studio now supports all Studio GraphQL APIs, with or without DataStore. [Learn more below](#datastore-and-graphql)**

Reviewer: Should this be a warning? Info might be better here.

Author: Done!

![datamodel](/images/console/datamodel.gif)

The experience for modeling data is the same in the sandbox and in Studio. With an AWS account, you will have real-time data synchronization capabilities and you will also be able to set authorization rules on your models. For more information, see [Authorization](/console/authz/authorization). All data models are provisioned in your account as AWS AppSync GraphQL APIs and Amazon DynamoDB tables. As with every feature of Studio, it can be further extended with the CLI.

### DataStore and GraphQL
When you save and deploy your data model, Amplify Studio generates a GraphQL API for you using AWS AppSync. By default, all data models built with Studio are provisioned as GraphQL APIs without DataStore - however, **Studio supports all Amplify GraphQL APIs, with and without DataStore**. 

> **[DataStore](/lib/datastore/getting-started)** is an on-device storage engine that automatically synchronizes data between your mobile and web apps and your database, enabling offline use. DataStore is automatically added to your API when you add a **Conflict Resolution** method. 

Reviewer: `automatically enabled for your API`?

> **[DataStore](/lib/datastore/getting-started)** is an on-device storage engine that automatically synchronizes data between your mobile and web apps and your database, enabling offline use. DataStore is automatically added to your API when you add a **Conflict Resolution** method. 

Reviewer 2: `DataStore is configured when you enable **Conflict Resolution** for your API`. was hoping if this made more sense as we configure the API to use data store additionally by enabling conflict resolution.

> **[DataStore](/lib/datastore/getting-started)** is an on-device storage engine that automatically synchronizes data between your mobile and web apps and your database, enabling offline use. DataStore is automatically added to your API when you add a **Conflict Resolution** method. 

Author: ```suggestion
**[DataStore](/lib/datastore/getting-started)** is an on-device storage engine that automatically synchronizes data between your mobile and web apps and your database, enabling offline use. DataStore is automatically enabled for your API when you add a **Conflict Resolution** method. 
```

<Callout warning>

> **IMPORTANT NOTE**: Changing your Conflict Resolution setting results is a **destructive data change**, and may result in inaccessible data. It's highly recommended that this setting only be changed on GraphQL APIs with **no data**. 

Reviewer: Really??? How so?

Author: Turning off Conflict Resolution deletes the columns (and the subsequent data) used in the conflict resolution strategy. These columns aren't needed if Conflict Resolution is disabled, so turning off Conflict Resolution is a **destructive** change, but a **non-breaking** data change. However, if the user then turns Conflict Resolution back on, that data will be inaccessible until they manually backfill the data in those re-added columns. 

This is a pretty nuanced difference that I didn't dive into in the docs because it's kinda complicated. I'm thinking maybe we should add a detailed paragraph explaining why the change is destructive below the callout.

Author: Confirmed that my info was wrong - updated this section accordingly

</Callout>

> To enable/disable Conflict Resolution, or to change your Conflict Resolution settings, take the following steps:

Reviewer: Should we add the CLI commands also? Maybe a tabbed interface with visual vs CLI?

2. Select **GraphQL API settings** at the top of the page
![screenshot of the Data tab, highlighting the GraphQL API settings link](/images/console/conflict-resolution1.png)
3. Toggle Conflict Resolution off or on, or change your Conflict Resolution strategy
![screenshot of the GraphQL API settings, highlighting the Conflict Resolution switch, and dropdown with options](/images/console/conflict-resolution2.png)
4. Return to the Data Modeling screen and select **Save and Deploy**

To configure your API using the Amplify CLI, run `amplify configure api` in your terminal and follow the prompts. 

## Limitations
Applications with Conflict Resolution disabled have some minor limitations:
- Sorting by column in [Data Manager](./content-management) is not supported
- [Figma-to-Code Collections](../uibuilder/collections/) do not support Pagination, and have limited support for Search
- [Forms with relationships](../formbuilder/special-inputs/) have some limitations when searching Autocomplete fields

To get access to these features, turn on DataStore and select a Conflict Resolution strategy.

### Unsupported directives and field-level parameters

> If a data model contains unsupported directives or field-level parameters, the **Visual editor** under Data modeling will not be available to use, and **Data manager** may have limited functionality. However, you can continue using other Amplify Studio features and use the Amplify CLI to make modifications to GraphQL schemas.

Reviewer 2: ```suggestion
If a data model contains unsupported directives or field-level parameters, the **Visual editor** under Data modeling, and **Data manager** may have limited functionality. However, you can continue using other Amplify Studio features and use the Amplify CLI to make modifications to GraphQL schemas.
```
